src/Og.php

```diff
@@ -17,6 +17,139 @@
+   * @param $entity_type
+   *   The entity type.
+   * @param $bundle
+   *   The bundle name.
+   */
+  public static function createField($field_name, $entity_type, $bundle) {
+    $og_field = static::fieldInfo($field_name)
+      ->setEntityType($entity_type)
+      ->setBundle($bundle);
+
+    if (!FieldStorageConfig::loadByName($entity_type, $field_name)) {
+      $og_field->fieldDefinition()->save();
+    }
+
+    // Allow overriding the field name.
+    // todo: ask if we need this.
+//    $og_field['field']['field_name'] = $field_name;
+//    if (empty($field)) {
+//      $og_field['field']->save();
+//    }
+
+    if (!FieldConfig::loadByName($entity_type, $bundle, $field_name)) {
+      $og_field->instanceDefinition()->save();
+      // Clear the entity property info cache, as OG fields might add different
+      // entity property info.
+      static::invalidateCache();
+    }
+
+    // Add the field to the form display manager.
+    $displayForm = EntityFormDisplay::load($entity_type . '.' . $bundle . '.default');
+    if (!$displayForm->getComponent($field_name) && $widgetDefinition = $og_field->widgetDefinition()) {
+      $displayForm->setComponent($field_name, $widgetDefinition);
+      $displayForm->save();
+    }
+
+    // Define the view mode for the field.
+    if ($fieldViewModes = $og_field->viewModesDefinition()) {
+      $prefix = $entity_type . '.' . $bundle . '.';
+      $viewModes = \Drupal::entityManager()->getStorage('entity_view_display')->loadMultiple(array_keys($fieldViewModes));
+
+      foreach ($viewModes as $key => $viewMode) {
+        $viewMode->setComponent($field_name, $fieldViewModes[$prefix . $key])->save();
+      }
+    }
+  }
+
+  /**
+   * Gets the groups an entity is associated with.
+   *
+   * @param $entity_type
+   *   The entity type. Defaults to 'user'
+   * @param $entity_id
+   *   (optional) The entity ID.
+   * @param $states
+   *   (optional) Array with the state to return. Defaults to active.
+   * @param $field_name
+   *   (optional) The field name associated with the group.
+   *
+   * @return array
+   *  An array with the group's entity type as the key, and array - keyed by
+   *  the OG membership ID and the group ID as the value. If nothing found,
+   *  then an empty array.
+   */
+  public static function getEntityGroups($entity_type = 'user', $entity_id = NULL, $states = [OG_STATE_ACTIVE], $field_name = NULL) {
+    $cache = &drupal_static(__FUNCTION__, array());
+
+    if ($entity_type == 'user' && empty($entity_id)) {
+      $account = \Drupal::currentUser()->getAccount();
+      $entity_id = $account->id();
+    }
+
+    // Get a string identifier of the states, so we can retrieve it from cache.
+    if ($states) {
+      sort($states);
+      $state_identifier = implode(':', $states);
+    }
+    else {
+      $state_identifier = FALSE;
+    }
+
+    $identifier = [
+      $entity_type,
+      $entity_id,
+      $state_identifier,
+      $field_name,
```

Comment:
no, I mean to keep as is. If `$field_name == NULL`, it's still a unique identifier for the query.